### ISSUE TITLE
Foundry v9 error: Uncaught Reference Error at QuickEncounter.js:541 : JournalEntryPage is not defined

### ISSUE THREAD

**Issue by spetzel2020** [spetzel2020/quick-encounters#109]
Reported by benik#5434.

Complete whiff by me here, not properly testing the new QE version 1.0.5 with Foundry v9.

Problem is here:
```
const openJournalEntry = ((candidateJournalEntry instanceof JournalEntry ) || (candidateJournalEntry instanceof JournalEntryPage))
                                    ? candidateJournalEntry : null;
```

This needs to incorporate a test for FoundryV10.

Also need to check for other places where `JournalEntryPage` is referenced without proper version checking.

**Comment by spetzel2020** [comment]
Other places where `JournalEntryPage` is referenced without version checking:
QuickEncounters.js:1017
```        //1.0.4k: Use parent (which is what is saved to the map) if this is JournalEntryPage
        const noteJournalEntry = (this.journalEntry instanceof JournalEntryPage) ? this.journalEntry.parent : this.journalEntry;
```

**Comment by spetzel2020** [comment]
Fixed in v1.0.6 (not yet released)

**Comment by spetzel2020** [comment]
Additional locations:
```
EncounterNote.js:219 Uncaught (in promise) ReferenceError: JournalEntryPage is not defined
[Detected 1 package: quick-encounters]
    at EncounterNote.getEncounterScene (EncounterNote.js:219:61)
    at QuickEncounter.displayQEDialog (QuickEncounter.js:1701:39)
    at onRenderJournalSheet (QuickEncounter.js:1628:28)
    at Hooks._call (foundry.js:294:14)
    at Hooks.call (foundry.js:279:33)
    at JournalSheet._render (foundry.js:3117:13)
    at async JournalSheet._render (foundry.js:3764:5)
    at async JournalSheet._render (foundry.js:49689:5)
```
```
QuickEncounter.js:1849 Uncaught (in promise) TypeError: journalEntry.pages?.keys is not iterable (cannot read property Symbol(Symbol.iterator))
[Detected 1 package: quick-encounters]
    at QuickEncounter.js:1849:56
    at Hooks._call (foundry.js:294:14)
    at Hooks.call (foundry.js:279:33)
    at JournalSheet.close (foundry.js:3466:13)
    at JournalSheet.close (foundry.js:4123:18)
    at async JournalSheet.close (foundry.js:4220:5)
```

**Comment by spetzel2020** [comment]
Fixed (again) in 1.0.7a

**Comment by spetzel2020** [comment]
Released in QE 1.0.7
